app/bundles/PageBundle/Model/PageModel.php

```diff
@@ -463,6 +474,20 @@ public function hitPage($page, Request $request, $code = '200', Lead $lead = nul
         // Get lead if required
         if (null == $lead) {
             $lead = $this->leadModel->getContactFromRequest($query, $this->trackByFingerprint);
+            // company
+            list($company, $leadAdded, $companyEntity) =  IdentifyCompanyHelper::identifyLeadsCompany($query, $lead, $this->companyModel);
+            if ($leadAdded) {
+                $lead->addCompanyChangeLogEntry('form', 'Identify Company', 'Lead added to the company, '.$company['companyname'], $company['id']);
+            } else {
+                $this->companyModel->setFieldValues($companyEntity, $query);
```

Comment:
I'm getting `Type error: Argument 1 passed to Mautic\LeadBundle\Model\CompanyModel::setFieldValues() must be an instance of Mautic\LeadBundle\Entity\Company, null given` on this line.

Reply:
Got it, fixed.